lua/lspconfig/vdmj.lua

```diff
@@ -0,0 +1,127 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+if vim.fn.has 'nvim-0.5.1' == 0 then
+  vim.notify('The VDMJ language server requires nvim > 0.5', vim.log.levels.ERROR)
+  return
+end
+
+local server_name = 'vdmj'
+
+local mavenrepo = util.path.join(vim.env.HOME, '/.m2/repository/com/fujitsu')
+
+local function get_jar_path(config, package, version)
+  return util.path.join(config.options.mavenrepo, package, version, package .. '-' .. version .. '.jar')
+end
+
+local function with_precision(version, is_high_precision)
+  return is_high_precision and version:gsub('([%d.]+)', '%1-P') or version
+end
+
+local function get_latest_installed_version(repo)
+  local path = util.path.join(repo, 'lsp')
+  local sort = vim.fn.sort
+
+  local subdirs = function(file)
+    local stat = vim.loop.fs_stat(util.path.join(path, file))
+    return stat.type == 'directory' and 1 or 0
+  end
+
+  local candidates = vim.fn.readdir(path, subdirs)
+  local sorted = sort(sort(candidates, 'l'), 'N')
+  return sorted[#sorted]
+end
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'java' },
+    filetypes = { 'vdmsl', 'vdmpp', 'vdmrt' },
+    root_dir = function(fname)
+      return util.find_git_ancestor(fname) or vim.fn.getcwd()
```

Comment:
remove cwd

Reply:
This was a conscious choice: VDM specifications are often enough distributed as-is, outside of a Git repository -- at the very least, that is the case at my workplace, where we have a mix of both.

I am happy to change this if there is a better solution, but `util.find_git_ancestor(fname)` (or looking for some sort of build files) is not enough.

Reply:
You should remove cwd, and suggest users add util.path.dirname(filename) in the instructions then.

cwd will lead to highly undesirable behavior (you open neovim in $HOME, open a subfolder the server indexes $HOME slowing down the entire computer. you open neovim in a subfolder of a project, open a file in a parent folder, neovim cannot index that file)

Reply:
While there are no "standard" files that would mark the root of a VDM specification, the VDMJ language server stores some (optional) configuration under the `.vscode` subfolder of the workspace root. Therefore I changed this to return either the Git ancestor, the VSCode ancestor, or the current file's directory, in order of preference.